h2/stream.py

```diff
@@ -131,8 +132,13 @@ def request_sent(self, previous_state):
         Fires when a request is sent.
         """
         self.client = True
-        self.headers_sent = True
-        return []
+        if not self.headers_sent:
+            self.headers_sent = True
+            event = _HeadersSent()
+        else:
+            event = _TrailersSent()
```

Comment:
Ah, no, they can, but it happens in `response_sent`. So I'm right, this branch is unhittable and we don't need it.